src/models/chen2021Fitness/Chen2021FitnessWidget.tsx

```diff
@@ -0,0 +1,381 @@
+
+const ContainerPropsSchema = SampleSelectorSchema;
+type ContainerProps = zod.infer<typeof SampleSelectorSchema>;
+
+const ResultsPropsSchema = zod.object({
+  request: Chen2021FitnessRequestSchema,
+});
+type ResultsProps = zod.infer<typeof ResultsPropsSchema>;
+
+type Chen2021FitnessRequest = zod.infer<typeof Chen2021FitnessRequestSchema>;
+type Chen2021FitnessResponse = zod.infer<typeof Chen2021FitnessResponseSchema>;
+
+const getData = async (
+  params: Chen2021FitnessRequest,
+  signal: AbortSignal
+): Promise<Chen2021FitnessResponse | null> => {
+  const mutationsString = params.mutations.join(',');
+  const url =
+    `/computed/model/chen2021Fitness` +
+    `?country=${params.country}` +
+    `&mutations=${mutationsString}` +
+    `&matchPercentage=${params.matchPercentage}` +
+    (params.samplingStrategy ? `&dataType=${params.samplingStrategy}` : '') +
+    `&alpha=${params.alpha}` +
+    `&generationTime=${params.generationTime}` +
+    `&reproductionNumberWildtype=${params.reproductionNumberWildtype}` +
+    `&plotStartDate=${params.plotStartDate.toISOString().substring(0, 10)}` +
+    `&plotEndDate=${params.plotEndDate.toISOString().substring(0, 10)}` +
+    `&initialWildtypeCases=${params.initialWildtypeCases}` +
+    `&initialVariantCases=${params.initialVariantCases}`;
+  const response = await get(url, signal);
+  if (response.status !== 200) {
+    // The computation might fail, for example, if some values go out-of-bound. The issue shall be addressed with the
+    // introduction of a better error handling and reporting on the server side.
+    return null;
+  }
+  const data = await response.json();
+  if (!data) {
+    return null;
+  }
+  return Chen2021FitnessResponseSchema.parse(data);
+};
+
+const Wrapper = styled.div`
+  background-color: #ffe0b6;
+  padding: 15px;
+`;
+
+const SectionHeader = styled.h5`
+  margin-top: 20px;
+`;
+
+const Chen2021FitnessContainer = ({
+  country,
+  mutations,
+  matchPercentage,
+  samplingStrategy,
+}: ContainerProps) => {
+  const [paramData, setParamData] = useState<Chen2021FitnessRequest>({
+    country,
+    mutations,
+    matchPercentage,
+    samplingStrategy,
+    alpha: 0.95,
+    generationTime: 4.8,
+    reproductionNumberWildtype: 1,
+    plotStartDate: new Date('2021-01-01'),
+    plotEndDate: new Date(Date.now() + 60 * 24 * 60 * 60 * 1000),
+    initialWildtypeCases: 1000,
+    initialVariantCases: 100,
+  });
+  const [formGenerationTime, setFormGenerationTime] = useState(paramData.generationTime.toString());
+  const [formReproductionNumberWildtype, setFormReproductionNumberWildtype] = useState(
+    paramData.reproductionNumberWildtype.toString()
+  );
+  const [formInitialVariantCases, setFormInitialVariantCases] = useState(
+    paramData.initialVariantCases.toString()
+  );
+  const [formInitialWildtypeCases, setFormInitialWildtypeCases] = useState(
+    paramData.initialWildtypeCases.toString()
+  );
+
+  useEffect(() => {
```

Comment:
I see how `useEffect` is useful in this specific case, so it's fine here. However in general `useMemo` is the right hook for this sort of thing.